### PR TITLE
Upload adult dataset

### PR DESCRIPTION
**Pull request by superp0tat0**
* Include auto generated docs from roxygen2
* Activate md for roxygen2
* Include the new datasets.

PS: Slightly changed the rda generation scripts so it could compress the train and test dataframe in a single adult.rda

### CODE REVIEW

File: R/adult.R [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#' * Rows contain NA in Workclass and Occupation has been removed.
+#'
+#' @source \url{https://archive.ics.uci.edu/ml/datasets/adult}
+#'
+#' @docType data
+#' @keywords data
+#' @examples
+#' data("adult", package = "mlr3fairness")
+#'
+
```
Comment (pfistfl) [inline R/adult.R:24]: no newlines after

File: R/adult.R [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+get_adult_task = function() {
```
Comment (pfistfl) [inline R/adult.R:26]: split this into adult_train and adult_test

File: R/adult.R [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  #Not sure if this need to be implemented. For now only add the documents
+}
```
Comment (pfistfl) [inline R/adult.R:28]: Add the tasks to `mlr_tasks`, see https://github.com/mlr-org/mlr3data/blob/main/R/kc_housing.R

File: R/adult.R [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+#' Train dataset contains 13 features and 30178 observations.
+#' Test dataset contains 13 features and 15315 observations.
+#' Target column is "Target".
+#'
+#' @section Pre-processing:
+#' * Country column has been removed since we expect them have no predictive power.
+#' * fnlwgt column has been removed since we expect them have no predictive power.
+#' * Rows contain NA in Workclass and Occupation has been removed.
+#'
+#' @source \url{https://archive.ics.uci.edu/ml/datasets/adult}
```
Comment (pfistfl) [inline R/adult.R:17]: Maybe check / hint at the LICENSE for the dataset

File: DESCRIPTION [hunk 5]
```diff
@@ -18,3 +18,4 @@ LazyData: true
 Depends:
     R (>= 3.1.0)
```
Comment (pfistfl) [inline DESCRIPTION:19]: This needs to be  >= 3.5.0

File: R/helper.R [hunk 6]
```diff
@@ -0,0 +1,8 @@
+as_backend = function(id) {
```
Comment (pfistfl) [inline R/helper.R:1]: create a `zzz.R` file and doc the required imports, see https://github.com/mlr-org/mlr3data/blob/0b1f8fa1b6775cb39ac67d099550ea4420b3d0af/R/zzz.R#L1

File: R/adult.R [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+#' year = "2017",
+#' title = "{UCI} Machine Learning Repository",
+#' url = "http://archive.ics.uci.edu/ml",
+#' institution = "University of California, Irvine, School of Information and Computer Sciences" }
+#'
+#' @docType data
+#' @keywords data
+#' @examples
+#' data("adult_test", package = "mlr3fairness")
+#' data("adult_train", package = "mlr3fairness")å
```
Comment (pfistfl) [inline R/adult.R:29]: trailing character

**Comment by pfistfl** [comment]
- [ ] Add unit tests for datasets. See `usethis::use_testthat()`  and https://github.com/mlr-org/mlr3data/blob/main/tests/test_data_sets.R

File: R/helper.R [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+  # Integrated from mlr3data subpackage
+  # this looks a bit funny, but is required because of strange
+  # behavior with roxygen2/pkgload
+  ee = new.env(hash = FALSE, parent = emptyenv())
+  data(list = id, package = "mlr3data", envir = ee)
```
Comment (pfistfl) [inline R/helper.R:6]: are you sure package needs to be "mlr3data" here?

**Comment by superp0tat0** [comment]
> * [ ]  Add unit tests for datasets. See `usethis::use_testthat()`  and https://github.com/mlr-org/mlr3data/blob/main/tests/test_data_sets.R
> * [ ]   Add the datasets to `mlr_tasks`, see https://github.com/mlr-org/mlr3data/blob/0b1f8fa1b6775cb39ac67d099550ea4420b3d0af/R/zzz.R

Not sure whether I need to add .onload and .onUnload for the same zzz.R. What its function?

**Comment by pfistfl** [comment]
See Mattermostt, yes add the onLoad \ onUnload

File: R/adult.R [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+#'
+#' @name adult
+#' @aliases adult_test
+#' @aliases adult_train
+#'
+#' @description
+#' Dataset used to predict whether income exceeds $50K/yr based on census data. Also known as "Census Income" dataset
+#' Train dataset contains 13 features and 30178 observations.
+#' Test dataset contains 13 features and 15315 observations.
+#' Target column is "Target".
```
Comment (pfistfl) [inline R/adult.R:11]: Can you write half a sentence more about what the two levels mean? It is already in L:8, but maybe not immediately clear.

File: R/adult.R [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+#'
+#' @section Pre-processing:
```
Comment (pfistfl) [inline R/adult.R:13]: Is there a reference as to where this kind of preprocessing was used? If yes, add; otherwise ignore this comment

**Comment by pfistfl** [comment]
Other than my very minor comments, this looks very good to me! Thanks a lot!
I'll merge once you addressed them.

File: R/adult.R (hunk 10 above)
Comment (superp0tat0) [reply R/adult.R:13]: I did not find a reference, but I followed some principles from examples in mlr3data like
https://github.com/mlr-org/mlr3data/blob/0b1f8fa1b6775cb39ac67d099550ea4420b3d0af/R/optdigits.R

**Comment by pfistfl** [comment]
Looks good, @mllg is it ok if I merge?

**Comment by mllg** [comment]
> Looks good, @mllg is it ok if I merge?

Sure.

**Pull request merged**